src/PostgREST/Error.hs

```diff
@@ -546,6 +577,45 @@ requiredTokenHeader = ("WWW-Authenticate", "Bearer")
 singularityError :: (Integral a) => a -> Error
 singularityError = SingularityError . toInteger
 
+-- For parsing byteString to JSON Object, used for allowing full response control
+data PgRaiseErrMessage = PgRaiseErrMessage {
+  getCode    :: Text,
+  getMessage :: Text,
+  getDetails :: Maybe Text,
+  getHint    :: Maybe Text
+}
+
+data PgRaiseErrDetails = PgRaiseErrDetails {
+  getStatus      :: Int,
+  getDescription :: Maybe Text,
+  getHeaders     :: Map Text Text
```

Comment:
```suggestion
  status      :: Int,
  description :: Maybe Text,
  headers     :: Map Text Text
```

Now that I see it, I think avoiding the "get" here is better as a data type.

Reply:
`description` to `statusText` would be better as well.

Reply:
@laurenceisla `status` and `headers` are already used by `PgrstError`, so I can't use those here otherwise variable name shadowing warning shows up.

Reply:
Thanks @wolfgangwalther. I'll change this in a future refactoring.